homeassistant/components/overkiz/coordinator.py

```diff
@@ -150,14 +104,69 @@ async def _get_devices(self) -> dict[str, Device]:
         _LOGGER.debug("Fetching all devices and state via /setup/devices")
         return {d.device_url: d for d in await self.client.get_devices(refresh=True)}
 
-    def places_to_area(self, place: Place) -> dict[str, str]:
-        """Convert places with sub_places to a flat dictionary."""
+    def _places_to_area(self, place: Place) -> dict[str, str]:
+        """Convert places with sub_places to a flat dictionary [placeoid, label])."""
         areas = {}
         if isinstance(place, Place):
             areas[place.oid] = place.label
 
         if isinstance(place.sub_places, list):
             for sub_place in place.sub_places:
-                areas.update(self.places_to_area(sub_place))
+                areas.update(self._places_to_area(sub_place))
 
         return areas
+
+    @EVENT_HANDLERS.register(EventName.DEVICE_AVAILABLE)
+    async def on_device_available(self, event: Event) -> None:
+        """Handle device available event."""
+        if event.device_url:
+            self.devices[event.device_url].available = True
+
+    @EVENT_HANDLERS.register((EventName.DEVICE_UNAVAILABLE, EventName.DEVICE_DISABLED))
+    async def on_device_unavailable_disabled(self, event: Event) -> None:
+        """Handle device unavailable / disabled event."""
+        if event.device_url:
+            self.devices[event.device_url].available = False
+
+    @EVENT_HANDLERS.register(EventName.DEVICE_STATE_CHANGED)
+    async def on_device_state_changed(self, event: Event) -> None:
+        """Handle device state changed event."""
+        if not event.device_url:
+            return
+
+        for state in event.device_states:
+            device = self.devices[event.device_url]
+            device.states[state.name] = state
```

Comment:
We moved the logic 1:1, this is the only function that has been changed. Last PR's we really made it complicated to pass mypy, but now it was possible to simplify this and keep 100% mypy strict.